test/common/upstream/local_address_selector_integration_test.cc

```diff
@@ -0,0 +1,192 @@
+#include <string>
+
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/upstream/upstream.h"
+
+#include "test/common/upstream/test_local_address_selector.h"
+#include "test/integration/http_protocol_integration.h"
+#include "test/mocks/upstream/cluster_info.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// Basic test that instantiates the ``DefaultLocalAddressSelector``.
+TEST_P(HttpProtocolIntegrationTest, Basic) {
+
+  // Specify both IPv4 and IPv6 source addresses, so the right one is picked
+  // based on IP version of upstream.
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    uint32_t const port_value = 1234;
+    auto bind_config = bootstrap.mutable_cluster_manager()->mutable_upstream_bind_config();
+    bind_config->mutable_source_address()->set_address("127.0.0.1");
+    bind_config->mutable_source_address()->set_port_value(port_value);
+    auto extra_source_address = bind_config->add_extra_source_addresses();
+    extra_source_address->mutable_address()->set_address("::1");
+    extra_source_address->mutable_address()->set_port_value(port_value);
+  });
+
+  initialize();
+
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "GET"}, {":path", "/test/long/url"}, {":scheme", "http"}, {":authority", "host"}};
+
+  auto response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
+
+  // Verify the proxied request was received upstream, as expected.
+  EXPECT_TRUE(upstream_request_->complete());
+  EXPECT_EQ(0U, upstream_request_->bodyLength());
```

Comment:
Is it possible to verify the upstream bind address is expected one by https://github.com/envoyproxy/envoy/blob/74bfcb0b591bb692db0b9b0700b4822ef8a6d777/test/integration/http_integration.h#L344 ?